### PR TITLE
testify: migrate test-infra to testify for distsql_test.go #28574

### LINKED ISSUE #28574: migrate test-infra to testify for `distsql_test.go`

### PR DESCRIPTION
**Pull request by znhh6018**
### What problem does this PR solve?

Issue Number: close #28574 

Problem Summary:

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/distsql_test.go [hunk 1]
```diff
@@ -200,58 +227,67 @@ func (s *testSuite3) TestIssue10178(c *C) {
 	tk.MustQuery("select * from t where a < 9223372036854775808").Check(testkit.Rows("9223372036854775807"))
 }
 
-func (s *testSuite3) TestInconsistentIndex(c *C) {
-	tk := testkit.NewTestKit(c, s.store)
+func TestInconsistentIndex(t *testing.T) {
+	t.Parallel()
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	dom, err := session.BootstrapSession(store)
```
Comment (hawkingrei) [inline executor/distsql_test.go:234]:
```
store, dom, clean := testkit.CreateMockStoreAndDomain(t)
defer clean()
```

**Comment by znhh6018** [comment]
image
if I don't remove "unlint:unsed" on checkGoroutineExists, goword fail.
After I removed "unlint:unsed" and commit , the check_dev fail....

File: executor/distsql_test.go [hunk 2]
```diff
@@ -21,23 +21,25 @@ import (
...
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/util/israce"
 	"github.com/pingcap/tidb/util/mock"
-	"github.com/pingcap/tidb/util/testkit"
+	"github.com/stretchr/testify/require"
+	"github.com/tikv/client-go/v2/testutils"
 )
 
-// nolint:unused
 func checkGoroutineExists(keyword string) bool {
```
Comment (hawkingrei) [inline executor/distsql_test.go:43]: it can remove because no one uses it.

Comment (znhh6018) [reply executor/distsql_test.go:43]: you mean removing the checkGoroutineExists function ?

Comment (hawkingrei) [reply executor/distsql_test.go:43]: Yes.

```
[2021-12-12T12:11:56.974Z] executor/distsql_test.go:43:6: func `checkGoroutineExists` is unused (unused)

[2021-12-12T12:11:56.974Z] func checkGoroutineExists(keyword string) bool {

[2021-12-12T12:11:56.974Z]      ^

```

Comment (znhh6018) [reply executor/distsql_test.go:43]: image
I found checkGoroutineExists is called in line 123...

Comment (hawkingrei) [reply executor/distsql_test.go:43]: I know. 

https://github.com/pingcap/tidb/blob/9cd0b49f7f5ca6d26e8e77e25ceda0afc970437b/executor/seqtest/seq_executor_serial_test.go#L750

there is the same function.

Comment (znhh6018) [reply executor/distsql_test.go:43]: how can I solve this , I tried to rename this function, it also prompted unused...

Comment (hawkingrei) [reply executor/distsql_test.go:43]: Try to delete ```executor/distsql_test.go:43:6 checkGoroutineExists ``` ?

Comment (hawkingrei) [reply executor/distsql_test.go:43]: make ```checkGoroutineExists``` public and delete the redundant.

Comment (znhh6018) [reply executor/distsql_test.go:43]: okay , let me try

Comment (hawkingrei) [reply executor/distsql_test.go:43]: package name ```executor_test``` can be replaced by ```executor```.

File: executor/distsql_test.go [hunk 3]
```diff
@@ -335,14 +363,18 @@ func (s *testSuite3) TestIndexLookUpStats(c *C) {
...
+	require.Equal(t, stats.Clone().String(), stats.String())
 	stats.Merge(stats.Clone())
-	c.Assert(stats.String(), Equals, "index_task: {total_time: 10s, fetch_handle: 4s, build: 2s, wait: 4s}, table_task: {total_time: 4s, num: 4, concurrency: 2}")
+	require.Equal(t, "index_task: {total_time: 10s, fetch_handle: 4s, build: 2s, wait: 4s}, table_task: {total_time: 4s, num: 4, concurrency: 2}", stats.String())
 }
 
-func (s *testSuite3) TestIndexLookUpGetResultChunk(c *C) {
-	tk := testkit.NewTestKit(c, s.store)
+func TestIndexLookUpGetResultChunk(t *testing.T) {
+	t.Parallel()
```
Comment (hawkingrei) [inline executor/distsql_test.go:373]: ```t.Parallel()``` need to be removed. it is a decision that we have made recently.

Comment (znhh6018) [reply executor/distsql_test.go]: sorry , not familiar with some git operations , things seemed went bad , just close and create a new PR............

**Pull request closed without merging**